encrypted-config-value/src/main/java/com/palantir/config/crypto/value/AesEncryptedValue.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.config.crypto.value;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.util.Arrays;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonDeserialize(as = ImmutableAesEncryptedValue.class)
+@JsonSerialize(as = ImmutableAesEncryptedValue.class)
+@JsonTypeName("AES")
+public abstract class AesEncryptedValue extends EncryptedValue {
```

Comment:
Should this be `AesGcmEncryptedValue` instead? If we have a non-GCM AES value, then I think it would need its own class since fields like `iv` and `tag` may not be applicable (and there may be other fields as well).

In all likelihood we probably won't add another AES cipher type so probably fine, but just flagging.